numba/help/inspector.py

```diff
@@ -0,0 +1,429 @@
+        self.print('<li>')
+        self.print('{}.<b>{}</b>'.format(
+            modname,
+            itemname,
+        ))
+        self.print(': <b>{}</b>'.format(typename))
+        self.print('<div><pre>', explained, '</pre></div>')
+
+        self.print("<ul>")
+        for tcls, source in sources.items():
+            if source:
+                self.print("<li>")
+                impl = source['name']
+                sig = source['sig']
+                filename = source['filename']
+                lines = source['lines']
+                self.print(
+                    "<p>defined by <b>{}</b>{} at {}:{}-{}</p>".format(
+                        self.escape(impl), self.escape(sig),
+                        self.escape(filename), lines[0], lines[1],
+                    ),
+                )
+                self.print('<p>{}</p>'.format(
+                    self.escape(source['docstring'] or '')
+                ))
+            else:
+                self.print("<li>{}".format(self.escape(str(tcls))))
+            self.print("</li>")
+        self.print("</ul>")
+        self.print('</li>')
+
+    def write_unsupported_item(self, modname, itemname):
+        self.print('<li>')
+        self.print('{}.<b>{}</b>: UNSUPPORTED'.format(
+            modname,
+            itemname,
+        ))
+        self.print('</li>')
+
+    def write_statistic(self, stats):
+        self.print('<p>{}</p>'.format(stats.describe()))
+
+
+class ReSTFormatter(Formatter):
+    """Formatter that output ReSTructured text format for Sphinx docs.
+    """
+    def escape(self, text):
+        return text
+
+    def title(self, text):
+        self.print(text)
+        self.print('=' * len(text))
+        self.print()
+
+    def begin_module_section(self, modname):
+        self.print(modname)
+        self.print('-' * len(modname))
+        self.print()
+
+    def end_module_section(self):
+        self.print()
+
+    def write_supported_item(self, modname, itemname, typename, explained, sources, alias):
+        self.print('.. function:: {}.{}'.format(modname, itemname))
+        self.print()
+
+        for tcls, source in sources.items():
+            if source:
+                impl = source['name']
+                sig = source['sig']
+                filename = source['filename']
+                lines = source['lines']
+                source_link = github_url.format(
+                    commit=commit,
+                    path=filename,
+                    firstline=lines[0],
+                    lastline=lines[1],
+                )
+                self.print(
+                    "   - defined by ``{}{}`` at `{}:{}-{} <{}>`_".format(
+                        impl, sig, filename, lines[0], lines[1], source_link,
+                    ),
+                )
+
+            else:
+                self.print("   - defined by ``{}``".format(str(tcls)))
+        self.print()
+
+        if alias:
+            self.print("   alias to: ``{}``".format(alias))
```

Comment:
I'm wondering if this ought to be declared first and lined up with the indented block. Locally it seems to render as e.g.:
```
numpy.max()

    * defined by <class 'numba.typing.npydecl.Numpy_redirect_max'>

alias to: numpy.amax
```
in firefox and chromimum (the indent is entirely ignored).

I think this might be a stylistic improvement:
```
numpy.max()

    Alias to: numpy.amax

    * defined by <class 'numba.typing.npydecl.Numpy_redirect_max'>
```

Reply:
The indent is only saying that "Alias..." is under the `numpy.max()` item.  But, I can't control it's indent further.  I moved the alias up to be just under the `numpy.max()` line.